platform/retarget.cpp

```diff
@@ -313,6 +355,10 @@ extern "C" int PREFIX(_read)(FILEHANDLE fh, unsigned char *buffer, unsigned int
 #endif
 #endif
         n = 1;
+#if defined MBED_CFG_DEBUG_OPTIONS_COVERAGE
+    } else if (coverage_report && fh == gcov_fd){
+        n = 0;
+#endif
```

Comment:
Nit: Same deal in the above block about spaces, there should be spaces after the `if ()`.

Ex. `if (blah){` should be `if (blah) {`.